packages/core/src/Dialog/Dialog.styled.tsx

```diff
@@ -0,0 +1,183 @@
+    borderRadius: '24px',
+  },
+  xl: {
+    width: 'calc(min(1280px, 100vw) - 2 * 16px)',
+    height: 'calc(100% - 2 * 24px)',
+    borderRadius: '24px',
+  },
+  full: {
+    width: '100%',
+    height: '100%',
+  },
+} as const
+
+const FloatingCloseButton: typeof CloseButton = styled(CloseButton)`
+  position: absolute;
+  top: ${(props) => -((props.dialogSize === 'full' || props.fullWidth ? 0 : 16) + 10)}px;
+  right: ${(props) => -((props.dialogSize === 'full' || props.fullWidth ? 0 : 16) + 10)}px;
+  margin: ${themeGet('space.3')};
+  padding: ${themeGet('space.2')};
+  background-color: ${themeGet('palette.background.lightest')};
+  box-shadow: ${themeGet('shadows.sm')};
+  &:hover {
+    background-color: ${themeGet('palette.background.lightest')};
+  }
+`
+
+const DialogOverlayWrapper = styled(motion.div)`
+  position: fixed;
+  inset: 0;
+  display: grid;
+  overflow-y: auto;
+  cursor: ${(props: IDialogProps) => (props.scrimDismiss ? 'pointer' : 'default')};
+  place-items: ${(props: IDialogProps) => (props.sheet ? 'end center' : 'center')};
+  background-color: ${(props: IDialogProps) => `rgba(0, 0, 0, ${props.scrimOpacity})`};
+`
+
+const DialogContentWrapper = styled(motion.div)`
+  position: relative;
+  cursor: initial;
+  background-color: ${themeGet('palette.background.lightest')};
+  width: ${(props: IDialogProps) => (props.fullWidth ? '100%' : sizeStyles[props.size]?.width)};
+  height: ${(props: IDialogProps) =>
+    props.sheet ? `calc(${sizeStyles[props.size]?.height} - 24px)` : sizeStyles[props.size]?.height};
+  margin: ${(props: IDialogProps) => (props.sheet ? '16px 0 0 0' : '32px 16px 64px 16px')};
+  border-radius: ${(props: IDialogProps) =>
+    props.sheet ? '24px 24px 0 0' : sizeStyles[props.size]?.borderRadius};
+  box-shadow: ${themeGet('shadows.overlay-lg')};
+`
+
+const DialogInnerContentWrapper = styled.div`
+  position: relative;
+  overflow: auto;
+  border-radius: ${(props: IDialogProps) =>
+    props.sheet ? '24px 24px 0 0' : sizeStyles[props.size]?.borderRadius};
+  border-top: ${(props: IDialogProps) =>
+    props.hugColor && `4px solid ${themeGet('palette.' + props.hugColor)(props)}`};
+`
+
+export const DialogOverlay = ({ scrimDismiss, scrimOpacity, sheet, children }: Partial<IDialogProps>) => {
+  return (
+    <Dialog.Portal>
+      <ThemeProvider>
+        <Dialog.Overlay asChild>
+          <DialogOverlayWrapper
+            scrimDismiss={scrimDismiss}
+            scrimOpacity={scrimOpacity}
+            sheet={sheet}
+            initial={{ opacity: 0 }}
+            animate={{ opacity: 1 }}
+          >
+            {children}
+          </DialogOverlayWrapper>
+        </Dialog.Overlay>
+      </ThemeProvider>
+    </Dialog.Portal>
+  )
+}
+
+export const DialogContent = ({
+  ariaDescription,
+  ariaTitle,
+  children,
+  fullWidth,
+  headerColorScheme,
+  headerContent,
+  headerIcon,
+  headerShowCloseButton,
+  hugColor,
+  scrimDismiss,
+  sheet,
+  showCloseButton,
```

Comment:
Would using the same prop for `showCloseButton` and `headerShowCloseButton` cause a functional difference?

Reply:
Yes, if both are enabled then it is possible to get an experience like this:

![image](https://github.com/priceline/design-system/assets/23641048/75497be4-821e-46d2-ae4c-27e68dea4dff)

The end user would only use whichever option fulfills the design spec. Both, when clicked, perform the same actions though!